textinput.go

```diff
@@ -104,7 +105,7 @@ func (p *textinputPlugin) handleSetEditingState(arguments interface{}) (reply in
 		return nil, errors.Wrap(err, "failed to decode json arguments for handleSetEditingState")
 	}
 
-	p.word = []rune(editingState.Text)
+	p.word = utf16.Encode([]rune(editingState.Text))
```

Comment:
utf16? Is that what Flutter uses?
I'd think utf8 would be the default for modern projects.... Especially with the "utf-8 everywhere" movement that's going on.
http://utf8everywhere.org/

Reply:
> utf16? Is that what Flutter uses?

Dart to be more specific!

Here is some readings:
* https://github.com/dart-lang/sdk/issues/28404
* https://github.com/flutter/flutter/blob/e0c63cd35e15e407a80dc44281cc392535fcce25/packages/flutter/lib/src/services/text_formatter.dart#L197-L204

A bug we encounter: https://github.com/go-flutter-desktop/go-flutter/issues/314

> I'd think utf8 would be the default for modern projects.... Especially with the "utf-8 everywhere" movement that's going on.

Well, the glfw callback returns a utf8 code point, we then convert it to utf16 at the embedder level and then send it to flutter.
On the outside, we support as input utf8-based code point :smile:... but on the inside.. it'sa Roller coaster casting character :parachute:

Reply:
In for the ride, choo choo! :roller_coaster: